cmd/vanitySubdomains.go

```diff
@@ -0,0 +1,95 @@
+package cmd
+
+import (
+	"errors"
+	"os"
+	"os/signal"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/supabase/cli/internal/vanity_subdomains/activate"
+	"github.com/supabase/cli/internal/vanity_subdomains/check"
+	"github.com/supabase/cli/internal/vanity_subdomains/delete"
+	"github.com/supabase/cli/internal/vanity_subdomains/get"
+)
+
+var (
+	vanityCmd = &cobra.Command{
+		GroupID: groupManagementAPI,
+		Use:     "vanity-subdomains",
+		Short:   "Manage vanity subdomains for Supabase projects",
+		Long: `Manage vanity subdomains for Supabase projects.
+
+Usage of vanity subdomains and custom domains is mutually exclusive.`,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if !experimental {
+				return errors.New("must set the --experimental flag to run this command")
+			}
+			return nil
```

Comment:
```suggestion
			return cmd.Root().PersistentPreRunE(cmd, args)
```

I just realised that cobra [PersistentPreRun is not chained](https://github.com/spf13/cobra/issues/252) from root cmd. So we need to call it explicitly when overriding